lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1546,8 +1560,10 @@ void FIRRTLLowering::run() {
         break;
       case LoweringFailure:
         // If lowering failed, don't remove *anything* we've lowered so far,
-        // there may be uses, and the pass will fail anyway.
+        // there may be uses, and the pass will fail anyway. Mark the failure in
+        // state.
         opsToRemove.clear();
+        circuitState.moduleLoweringFailed = true;
```

Comment:
Since this happens in a `parallelForEachN`, I added a bit of state to track if any calls failed. Not sure if there is a better way to bubble up failure from `parallelForEachN`.

Reply:
`failableParallelForeachN` may be what you're looking for.